dpdata/lammps/dump.py

```diff
@@ -51,29 +51,26 @@ def get_natoms_vec(lines) :
     assert (sum(natoms_vec) == get_natoms(lines))
     return natoms_vec
 
-def get_posi(lines) :
-    blk, head = _get_block(lines, 'ATOMS')
-    keys = head.split()
-    id_idx = keys.index('id') - 2
-    xidx = keys.index('x') - 2
-    yidx = keys.index('y') - 2
-    zidx = keys.index('z') - 2
-    sel = (xidx, yidx, zidx)
-    posis = []
-    for ii in blk :
-        words = ii.split()
-        posis.append([float(words[id_idx]), float(words[xidx]), float(words[yidx]), float(words[zidx])])
-    posis.sort()
-    posis = np.array(posis)     
-    return posis[:,1:4]
+def get_coordtype_and_scalefactor(keys):
+    # 4 types in total,with different scaling factor
+    key_pc=['x','y','z'] # plain cartesian, sf = 1
+    key_uc=['xu','yu','zu'] # unwraped cartesian, sf = 1
+    key_s=['xs','ys','zs'] # scaled by lattice parameter, sf = lattice parameter
+    key_su = ['xsu','ysu','zsu'] #scaled and unfolded,sf = lattice parameter
+    lmp_coor_type = [key_pc,key_uc,key_su,key_sf]
+    sf = [0,0,1,1]
+    for k in range(4):
+        if all(i in keys for i in lmp_coor_type(k)):
+            return lmp_coor_type(k),sf[k]
```

Comment:
`lmp_coor_type(k)` should be `lmp_coor_type[k]`... Please test the program locally before opening the PR. Also, some new tests for new situations are required.